### PR TITLE
Don't add accessibility modifiers to partial classes in new documents

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/55703

Unfortunately we don't have enough information to go on when processing new documents to do anything smarter here :(

### CODE REVIEW

**Review by sharwell — APPROVED** [review]

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs [hunk 1]
```diff
@@ -548,5 +548,38 @@ internal struct S1 { }
                 LanguageVersion = LanguageVersion.CSharp10,
             }.RunAsync();
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddAccessibilityModifiers)]
+        [WorkItem(55703, "https://github.com/dotnet/roslyn/issues/55703")]
+        public async Task TestPartial_WithExistingModifier()
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs:554]: do we need to do something similar for VB?

File: src/Features/CSharp/Portable/Formatting/CSharpAccessibilityModifiersNewDocumentFormattingProvider.cs [hunk 2]
```diff
@@ -47,6 +47,25 @@ public async Task<Document> FormatNewDocumentAsync(Document document, Document?
...
+                //    public partial class C { }
+                //
+                // When we see File1, we don't know about File2, so would add an internal modifier, which would result in a compile
+                // error.
+                var modifiers = syntaxFacts.GetModifiers(declaration);
+                syntaxFacts.GetAccessibilityAndModifiers(modifiers, out _, out var declarationModifiers, out _);
+                if (declarationModifiers.IsPartial)
+                {
+                    continue;
+                }
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Formatting/CSharpAccessibilityModifiersNewDocumentFormattingProvider.cs:67]:
```suggestion
                    continue;
```

**Review by CyrusNajmabadi — APPROVED** [review]
c# side lgtm.  but can we validate if we have an issue with VB?

File: src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs (hunk 1 above)
Comment (davidwengier) [reply src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs:554]: This test was just a regression check to ensure the analyzer didn't have the same issue as the new document handler, and it doesn't. The new document handler doesn't have a VB implementation (and as Dustin pointed out in the issue, VB doesn't require the partial modifier in both places so it probably never should)

**Comment by davidwengier** [comment]
> validate if we have an issue with VB?

Mentioned above, but for visibility, adjusting accessibility modifiers in new documents only happens for C# files.

File: src/Features/CSharp/Portable/Formatting/CSharpAccessibilityModifiersNewDocumentFormattingProvider.cs (hunk 2 above)
Comment (davidwengier) [reply src/Features/CSharp/Portable/Formatting/CSharpAccessibilityModifiersNewDocumentFormattingProvider.cs:67]: opinion

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Formatting/CSharpAccessibilityModifiersNewDocumentFormattingProvider.cs:67]: and the rest of the code in this method :)

**Pull request merged**